code/go/internal/validator/spec.go

```diff
@@ -140,6 +140,7 @@ func (s Spec) rules(pkgType string, rootSpec spectypes.ItemSpec) validationRules
 		{fn: semantic.ValidateRoutingRulesAndDataset, types: []string{"integration"}, since: semver.MustParse("2.9.0")},
 		{fn: semantic.ValidateKibanaNoDanglingObjectIDs, since: semver.MustParse("3.0.0")},
 		{fn: semantic.ValidateKibanaFilterPresent, since: semver.MustParse("3.0.0")},
+		{fn: semantic.ValidateKibanaNoLegacyVisualizations, types: []string{"integration"}, since: semver.MustParse("3.0.0")},
```

Comment:
Should this rule be limited to type "integration"?

Reply:
That should we ok, since input packages are not allowed to have kibana assets, just integration packages.
https://github.com/elastic/package-spec/blob/49120aea627a1652823a7f344ba3d1c9b029fd5a/spec/input/spec.yml